### ISSUE TITLE
[Bug] xref links to .NET classes don't resolve unless the class is also referenced via cref from an XML comment

### ISSUE THREAD

**Issue by tom-weiland** [dotnet/docfx#9480]
**Describe the bug**
Markdown articles are unable to resolve xref links to .NET classes unless the class is also referenced via cref from an XML comment within the project's code. (The same goes for .NET namespaces, enums, class members, etc.)

**To Reproduce**
1. Create a markdown article in the `manual` folder.
2. In the article, add an xref link to a .NET class, for example `<xref:System.DateTime>`. NOTE: there must be no references to the .NET class from XML comments project's code at this point (`<see cref="DateTime"/>` in our example).
3. Run `docfx docfx.json --serve` and notice the warning: `warning UidNotFound: 1 invalid cross reference(s) "<xref:System.DateTime>".`
4. View the created article in your browser and notice that the xref link did not resolve and is rendered as plaintext:
> \<xref:System.DateTime\>

5. In the project that you're building the documentation for, add an XML comment with a cref reference to the .NET class:
```cs
/// <summary><see cref="System.DateTime"/></summary>
public class SomeClass { }
```

6. Build the docs again.
7. Notice that the build warning is gone and the markdown article's xref link has resolved properly.
8. Remove the XML comment with the cref reference and rebuild the docs.
9. Notice that the build warning is back and the markdown article's xref link once again does not resolve properly.

**Expected behavior**
The xref link in the markdown article should resolve properly even if the thing being linked to is not cref referenced from any XML comments in the project code.

**Context:**
- OS: Windows 10
- Docfx version: v2.70.1

**Additional context**
I originally posted a discussions question about this (#9476). Discussion #9284 seems like it might be about the same issue.

**Comment by yufeih** [comment]
Try set `xref` to `https://learn.microsoft.com/en-us/dotnet/.xrefmap.json` to use the `xref` syntax against BCL types in markdown:

```json
{
  "build": {
    "xref": [
      "https://learn.microsoft.com/en-us/dotnet/.xrefmap.json"
    ]
  }
}
```

**Comment by tom-weiland** [comment]
Yep, adding that did the trick. Thanks for the quick response!

When I saw that the `xrefService` config option was being deprecated, I didn't realize that `xref` basically replaced it (I saw it in the config reference but wasn't entirely sure what it did or how to use it properly) and I guess I assumed that xrefs were now resolved "automagically" somehow 😅

**Comment by yufeih** [comment]
Yes, I need to update docs to reflect that. Xrefs in API reference docs did "automatically" resolve, but not for markdowns.
